src/_pytest/_io/saferepr.py

```diff
@@ -104,10 +106,14 @@ def saferepr(obj: object, maxsize: Optional[int] = DEFAULT_REPR_MAX_SIZE) -> str
     This function is a wrapper around the Repr/reprlib functionality of the
     stdlib.
     """
+
+    if use_ascii:
+        obj = ascii(obj)
+
     return SafeRepr(maxsize).repr(obj)
```

Comment:
I think we need to push this further down into `SafeRepr`; as-is I think we'll take the (safe) repr of the result of `ascii()` and end up with nested quote-marks.

Reply:
Oh, I hadn't noticed that the ascii() return is already safe, so calling repr is not required :smile: 

How about now?